packages/vscode-extension/src/devices/AndroidEmulatorDevice.ts

```diff
@@ -203,7 +165,44 @@ export class AndroidEmulatorDevice extends DeviceBase {
       const long = areCoordinatesToCloseToZero ? "0.00001" : settings.location.longitude.toString();
       await exec(ADB_PATH, ["-s", this.serial!, "emu", "geo", "fix", long, lat]);
     }
-    return shouldRestart;
+
+    // if the boot settings were changed, we need to restart the emulator
+    return settingsChanged;
+  }
+
+  /**
+   * This method updates the configuration settings for the emulator.
+   * It reads the config.ini file, applies the setting appliers, and writes the changes back to the file.
+   * @param settingAppliers - The setting appliers to apply.
+   * @returns if any changes were made.
+   */
+  private async tryUpdateConfigurationSettings(settingAppliers: SettingApplier[]) {
+    const avdDirectory = getAvdDirectoryLocation(this.avdId);
+    const configIni = path.join(avdDirectory, `${this.avdId}.avd`, "config.ini");
+    const configIniContent = await fs.promises.readFile(configIni, "utf-8");
+    const initialConfigLines = configIniContent.split("\n");
+    let configLines = initialConfigLines;
+
+    let bootSettingsChanged = false;
+    for (const applier of settingAppliers) {
+      if (await applier.shouldUpdate(configLines)) {
+        const { bootChanged, updatedConfigLines } = await applier.updateSettings(configLines);
+
+        if (bootChanged || updatedConfigLines !== undefined) {
+          bootSettingsChanged = true;
+        }
+        if (updatedConfigLines !== undefined) {
+          configLines = updatedConfigLines;
+        }
+      }
+    }
+
+    const finalizedConfig = configLines.join("\n");
+    if (bootSettingsChanged && finalizedConfig !== configIniContent) {
```

Comment:
the semantics of this new abstraction still confuses me. It'd seem more natural that the boot flag indicates whether we need to reboot after change, and the updatedConfig would indicate when we need to update the config. But this doesn't seem to be the case as we need to have both boot flag and the config flag set in order to apply the config updates.

Then, the implementation can actually return boot:false but also the list of updates, which are going to be discarded anyways.